modules/Emails/Email.php

```diff
@@ -3383,4 +3383,208 @@ public function revertFieldNullable($field)
             unset($this->modifiedFieldDefs[$field]);
             }
     	}
+
+    /**
+     * Uses the $_REQUEST to populate the fields of an Email SugarBean.
+     *
+     * @param Email $bean
+     * @return bool|Email|SugarBean
+     */
+    public function populateBeanFromRequest(Email $bean) {
+        if (empty($bean)) {
+            $bean = BeanFactory::getBean('Emails');
+        }
+
+        if(isset($_REQUEST['id'])) {
+            $bean = $bean->retrieve($_REQUEST['id']);
+        }
+
+
+        foreach ($_REQUEST as $fieldName => $field) {
+            if (array_key_exists($fieldName, $bean->field_defs)) {
+                $bean->$fieldName = $field;
+            }
+        }
+
+
+        $old = array('&lt;', '&gt;');
+        $new = array('<', '>');
+
+        if (isset($_REQUEST['from_addr']) && $_REQUEST['from_addr'] != $_REQUEST['from_addr_name'] . ' &lt;' . $_REQUEST['from_addr_email'] . '&gt;') {
+            if (false === strpos($_REQUEST['from_addr'], '&lt;')) { // we have an email only?
+                $bean->from_addr = $_REQUEST['from_addr'];
+                $bean->from_name = '';
+                $bean->reply_to_addr = $bean->from_addr;
+                $bean->reply_to_name = $bean->from_name;
+            } else { // we have a compound string
+                $newFromAddr = str_replace($old, $new, $_REQUEST['from_addr']);
+                $bean->from_addr = substr($newFromAddr, (1 + strpos($newFromAddr, '<')),
+                    (strpos($newFromAddr, '>') - strpos($newFromAddr, '<')) - 1);
+                $bean->from_name = substr($newFromAddr, 0, (strpos($newFromAddr, '<') - 1));
+                $bean->reply_to_addr = $bean->from_addr;
+                $bean->reply_to_name = $bean->from_name;
+            }
+        } elseif (!empty($_REQUEST['from_addr_email']) && isset($_REQUEST['from_addr_email'])) {
+            $bean->from_addr = $_REQUEST['from_addr_email'];
+            $bean->from_name = $_REQUEST['from_addr_name'];
+        } else {
+            $bean->from_addr = $bean->getSystemDefaultEmail();
+            $bean->reply_to_addr = $bean->from_addr['email'];
+            $bean->reply_to_name = $bean->from_addr['name'];
+        }
+
+
+        if (empty($bean->to_addrs)) {
+            if (!empty($_REQUEST['to_addrs_names'])) {
+                $bean->to_addrs_names = htmlspecialchars_decode($_REQUEST['to_addrs_names']);
+            }
+
+            if (!empty($bean->to_addrs_names)) {
+                $bean->to_addrs = htmlspecialchars_decode($bean->to_addrs_names);
+            }
+        }
+
+
+        $toEmailAddresses = preg_split('/[,;]/', $bean->to_addrs, null, PREG_SPLIT_NO_EMPTY);
+        $bean->to_addr_arr = array();
+        foreach ($toEmailAddresses as $ea => $address) {
+
+            $email = '';
+            $display = '';
```

Comment:
email and display are used see line 3507. these are temp variables used to work strip out the name from the email address (Joe Blogs <Joe@example.com>